src/types/inc/UTF8OutPipeReader.hpp

```diff
@@ -46,23 +46,23 @@ class UTF8OutPipeReader final
     };
 
     // array of bitmasks
-    constexpr const static BYTE _cmpMasks[]{
+    constexpr const static std::array<BYTE, 4> _cmpMasks{
         0, // unused
         _Utf8BitMasks::MaskContinuationByte,
         _Utf8BitMasks::MaskLeadByteTwoByteSequence,
         _Utf8BitMasks::MaskLeadByteThreeByteSequence,
     };
 
     // array of values for the comparisons
-    constexpr const static BYTE _cmpOperands[]{
+    constexpr const static std::array<BYTE, 4> _cmpOperands{
         0, // unused
         _Utf8BitMasks::IsAsciiByte, // intentionally conflicts with MaskContinuationByte
         _Utf8BitMasks::IsLeadByteTwoByteSequence,
         _Utf8BitMasks::IsLeadByteThreeByteSequence,
     };
 
     HANDLE _outPipe; // non-owning reference to a pipe.
-    BYTE _buffer[4096]{ 0 }; // buffer for the chunk read
-    BYTE _utf8Partials[4]{ 0 }; // buffer for code units of a partial UTF-8 code point that have to be cached
+    std::array<BYTE, 4096> _buffer; // buffer for the chunk read.
+    std::array<BYTE, 4> _utf8Partials; // buffer for code units of a partial UTF-8 code point that have to be cached

```

Comment:
Maybe use braces for the initialization here instead of the `fill()` method in the constructor.
`std::array<BYTE, 4096> _buffer{}; // buffer for the chunk read.`
For _utf8Partials respectively.